src/main/kotlin/compiler/parser/Parser.kt

```diff
@@ -655,14 +655,15 @@ class Parser(inputTokens: List<Token>) {
     // Parse lambda: { var, var -> ... } or { ... }
     private fun pLambdaFun(): N_EXPR? {
         consume(T_BRACE_OPEN)?.also {
-            var done = false
             val args = mutableListOf<N_IDENTIFIER>()
-            while (!done) {
-                done = true
+            while (nextAre(T_IDENTIFIER, T_COMMA)) {
+                consume(T_IDENTIFIER)?.also { args.add(N_IDENTIFIER(it.string)) }
+                consume(T_COMMA)
+            }
+            while (nextAre(T_IDENTIFIER, T_ARROW)) {
```

Comment:
Bugfix for parsing lambdas -- the argless form { ... } could not be parsed if it started with an identifier, for instance { x > 10 } -- the parser saw this as an incomplete arg declaration.